lib/Controller/DelegationController.php

```diff
@@ -0,0 +1,111 @@
+<?php
+namespace OCA\GroupFolders\Controller;
+
+use OCP\IConfig;
+use OCP\AppFramework\Http\JSONResponse;
+use OCP\AppFramework\Controller;
+use OCP\IGroup;
+use OCP\IGroupManager;
+use OCP\IRequest;
+use OCP\IUserManager;
+
+class DelegationController extends Controller {
+
+	/** @var IConfig */
+	private $config;
+
+	/** @var IGroupManager */
+	private $groupManager;
+
+	/** @var IUserManager */
+	private $userManager;
+
+	public function __construct($AppName,
+			IConfig $config,
+			IGroupManager $groupManager,
+			IRequest $request,
+			IUserManager $userManager) {
+		parent::__construct($AppName, $request);
+		$this->config = $config;
+		$this->groupManager = $groupManager;
+		$this->userManager = $userManager;
+	}
+
+	/**
+	 * Returns the list of all groups
+	 *
+	 * @NoAdminRequired
+	 * @NoCSRFRequired
```

Comment:
it should check for a CSRF token

Reply:
I thought it was for create and edit method only and not for getters method ?

Reply:
Yes, but as there is no reason to not check it here i'd prefer to just stick with the default ;)

Reply:
done